libtransmission/announcer.cc

```diff
@@ -880,19 +877,29 @@ void on_announce_error(tr_tier* tier, char const* err, tr_announce_event e)
     }
     else
     {
-        /* schedule a reannounce */
-        auto const interval = current_tracker->getRetryInterval();
-        tr_logAddWarnTier(
-            tier,
-            fmt::format(
-                tr_ngettext(
-                    "Announce error: {error} (Retrying in {count} second)",
-                    "Announce error: {error} (Retrying in {count} seconds)",
-                    interval),
-                fmt::arg("error", err),
-                fmt::arg("count", interval))
-                .append(fmt::format(" ({})", announce_url)));
-        tier_announce_event_push(tier, e, tr_time() + interval);
+        /* switch to the next tracker */
+        current_tracker = tier->useNextTracker();
```

Comment:
Question: why the change? Why, if it's a registered error we move to next tracker but if it's an unregistered error we don't?

Reply:
my bad. I thought the result was used only in that else branch, but useNextTracker() has side-effects too. I'll move it back.